datafusion/common/src/file_options/parquet_writer.rs

```diff
@@ -52,92 +53,43 @@ impl ParquetWriterOptions {
-            statistics_enabled,
-            max_statistics_size,
-            bloom_filter_fpp,
-            bloom_filter_ndv,
-            //  below is not part of ParquetWriterOptions
-            enable_page_index: _,
-            pruning: _,
-            skip_metadata: _,
-            metadata_size_hint: _,
-            pushdown_filters: _,
-            reorder_filters: _,
-            allow_single_file_parallelism: _,
-            maximum_parallel_row_group_writers: _,
-            maximum_buffered_record_batches_per_stream: _,
-            bloom_filter_on_read: _,
-        } = &parquet_options.global;
-
-        let key_value_metadata = if !parquet_options.key_value_metadata.is_empty() {
-            Some(
-                parquet_options
-                    .key_value_metadata
-                    .clone()
-                    .drain()
-                    .map(|(key, value)| KeyValue { key, value })
-                    .collect::<Vec<_>>(),
-            )
-        } else {
-            None
-        };
-
-        let mut builder = WriterProperties::builder()
-            .set_data_page_size_limit(*data_pagesize_limit)
-            .set_write_batch_size(*write_batch_size)
-            .set_writer_version(parse_version_string(writer_version.as_str())?)
-            .set_dictionary_page_size_limit(*dictionary_page_size_limit)
-            .set_max_row_group_size(*max_row_group_size)
-            .set_created_by(created_by.clone())
-            .set_column_index_truncate_length(*column_index_truncate_length)
-            .set_data_page_row_count_limit(*data_page_row_count_limit)
-            .set_bloom_filter_enabled(*bloom_filter_on_write)
-            .set_key_value_metadata(key_value_metadata);
-
-        if let Some(encoding) = &encoding {
-            builder = builder.set_encoding(parse_encoding_string(encoding)?);
-        }
-
-        if let Some(enabled) = dictionary_enabled {
-            builder = builder.set_dictionary_enabled(*enabled);
-        }
-
-        if let Some(compression) = &compression {
-            builder = builder.set_compression(parse_compression_string(compression)?);
-        }
-
-        if let Some(statistics) = &statistics_enabled {
-            builder =
-                builder.set_statistics_enabled(parse_statistics_string(statistics)?);
-        }
-
-        if let Some(size) = max_statistics_size {
-            builder = builder.set_max_statistics_size(*size);
-        }
+    fn try_from(parquet_table_options: &TableParquetOptions) -> Result<Self> {
+        // ParquetWriterOptions will have defaults for the remaining fields (e.g. sorting_columns)
+        Ok(ParquetWriterOptions {
+            writer_options: WriterPropertiesBuilder::try_from(parquet_table_options)?
+                .build(),
+        })
+    }
+}
 
-        if let Some(fpp) = bloom_filter_fpp {
-            builder = builder.set_bloom_filter_fpp(*fpp);
-        }
+impl TryFrom<&TableParquetOptions> for WriterPropertiesBuilder {
```

Comment:
I am not sure how important it is, but this API requires copying all the relevant fields / kv metadata, etc even if the caller had a owned copy

One thing that might make it more explicit about when copying was occuring was to change this to

```rust
impl TryFrom<TableParquetOptions> for WriterPropertiesBuilder {
```

And change `writer_props_from_global_opts` to also take self by value `self` rather than `&self`

Reply:
BTW I am not sure if this is worth doing, I just wanted to bring it up as an option

Reply:
Update here is @wiedld  mentioned to me that the `TableParquetOptions` are kept and a new `WrtierProperties` are created for each query so there isn't anything to be gained with this proposed change

Reply:
We chatted offline. The session config (`TableParquetOptions`) is re-used repeatedly in an execution, versus deriving the `WriterPropertiesBuilder` per each use of an arrow writer. Which means that we pass by reference, or clone the session config each time.

We decide it's ok as is. 👍🏼